common/src/main/java/com/wynntils/features/ui/CustomLoadingScreenFeature.java

```diff
@@ -41,6 +42,12 @@ public void onScreenOpenPre(ScreenOpenedEvent.Pre event) {
             baseConnectScreen = null;
         }
 
+        // Enable the custom loading screen for world transfers
+        if (MixinHelper.onWynncraft() && event.getScreen() instanceof ReceivingLevelScreen && loadingScreen == null) {
```

Comment:
As the name implies, MixinHelper should only be used in Mixins.

Reply:
There also already is a ReceivingLevelScreen case in this method, so perhaps instead you should create the screen in the `INTERIM` case for `onWorldStateChange` if there is no current loading screen

Reply:
Thanks! I implemented your suggestions.